File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        vector<int> the;
        
        for (int i = 0; i < nums.size(); i++) {
            for (int j = 0; j < nums.size(); j++) {
                if (!(i == j)) {
                    if (nums[i] + nums[j] == target) {
                        the.push_back(i);
                        the.push_back(j);
                        return the;
                    }
                }
            }
        }
        return nums;
    }
};

File: 0001-two-sum/0001-two-sum.scala
import scala.collection.mutable.Map

object Solution {
    def twoSum(nums: Array[Int], target: Int): Array[Int] = {
        val indexMap: Map[Int, Int] = Map()

        def recurse(currIndex: Int): Array[Int] = {
            if (currIndex >= nums.size) Array()

            if (indexMap.contains(target - nums(currIndex)) == true) return Array(currIndex, indexMap(target - nums(currIndex)))
            else indexMap += (nums(currIndex) -> currIndex)

            recurse(currIndex + 1)
        }
        
        recurse(0)
    }
}